datafusion/physical-plan/src/repartition/distributor_channels.rs

```diff
@@ -53,19 +57,20 @@ pub fn channels<T>(
 ) -> (Vec<DistributionSender<T>>, Vec<DistributionReceiver<T>>) {
     let channels = (0..n)
         .map(|id| {
-            Arc::new(Mutex::new(Channel {
-                data: VecDeque::default(),
-                n_senders: 1,
-                recv_alive: true,
-                recv_wakers: Vec::default(),
+            Arc::new(Channel {
```

Comment:
It would help me with readability / encapsulation to make this `Channel::new()` or `Channel::default()` so it is easier to follow the locking logic